protocol/clproto_cpp/CMakeLists.txt

```diff
@@ -27,22 +27,19 @@ add_custom_target(generate_proto_bindings COMMAND make cpp_bindings
 
 file(GLOB_RECURSE GENERATED_PROTO_BINDINGS "${PROTOBUF_BINDINGDS_DIR}/*.pb.cc" "${PROTOBUF_BINDINGDS_DIR}/*.pb.h")
 
-add_library(${PROJECT_NAME}_bindings ${GENERATED_PROTO_BINDINGS})
+add_library(${PROJECT_NAME}_bindings STATIC ${GENERATED_PROTO_BINDINGS})
```

Comment:
What type of library is this now (without SHARED or STATIC)?
`If no type is given explicitly the type is STATIC or SHARED based on whether the current value of the variable BUILD_SHARED_LIBS is ON. For SHARED and MODULE libraries the POSITION_INDEPENDENT_CODE target property is set to ON automatically.

Reply:
Well, as your reference indicates, it was defaulting to static before without any additional options. This also means position independent code (the ability for a shared library to compile a static library as an object file) was off.
Now it's explicitly static, with PIC marked on.